src/site/content/en/fast/serve-images-webp/index.md

```diff
@@ -156,6 +156,30 @@ ignore the tags it doesn't support. Thus, it only "sees" the
 - The resource specified by the `<img>` tag should be in a universally supported format (e.g. JPEG), so it can be used as a fallback.
 {% endAside %}
 
+### Reading the HTTP `Accept` header
+
+If you have an application back end or web server that allows you to rewrite requests, you can read the value of the [HTTP `Accept` header](https://developer.mozilla.org/docs/Web/HTTP/Headers/Accept), which will advertise what alternative image formats are supported:
+
+```http
+Accept: image/webp,image/svg+xml,image/*,*/*;q=0.8
+```
+
+Reading this request header and rewriting the response based on its contents has the benefit of simplifying your image markup. `<picture>` markup can get rather long with many sources. Below is an Apache `mod_rewrite` rule that can serve WebP alternates:
+
+```apacheconf
+RewriteEngine On
+RewriteCond %{HTTP:Accept} image/webp [NC]
+RewriteCond %{HTTP:Content-Disposition} !attachment [NC]
+RewriteCond %{DOCUMENT_ROOT}/$1.webp -f [NC]
+RewriteRule (.+)\.(png|jpe?g)$ $1.webp [T=image/webp,L]
+```
+
+The rewrite rule above will look for a WebP version of any requested JPEG or PNG image. If a WebP alternate is found, it will be served with the proper `Content-Type`  header. This will allow you to use image markup similar to the following with automatic WebP support:
+
+```html
+<img src="flower-320w.jpg" srcset="flower-320w.jpg 320w, flower-640w.jpg 640w, flower-960w.jpg 960w">
+```
```

Comment:
Personally I don't like serving webp images with a .jpg extension but that's just me :-)
But I do agree it can help reduce the markup considerably.

At the very least though, we should add a `vary` header to these responses, so they are cached correctly at any CDN or other intermediary cache.

Reply:
Agreed on `Vary`. Just need to look up the syntax for setting the `Vary` header in `mod_rewrite`.